net/socks5/conn.go

```diff
@@ -0,0 +1,151 @@
+package socks5
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"strconv"
+)
+
+type DialContext func(ctx context.Context, network, addr string) (net.Conn, error)
+
+// Conn represents a SOCKS5 connection for client to reach
+// server. The struct is filled by each of the internal methods
```

Comment:
> "The struct..."

This sentence should be moved into the struct itself; this doc comment should be aimed at users of the package, and should not discuss implementation details.